### PR TITLE
feat: add healthcheck for language server client

### LINKED ISSUE #15567: :checkhealth should check LSP log level
### Feature already in Vim?

_No response_

### Feature description

I found the lsp.log file in cache directory has over 700M file size. It would be great if its size gets restricted.

### PR DESCRIPTION
**Pull request by mjlbach**
Current example, open to change:

```
health#lsp#check                                                                                                               
========================================================================                                                       
## Checking language server client configuration                                                                               
  - INFO: LSP log level : DEBUG                                                                                                
  - WARNING: Log level DEBUG will cause degraded performance and high disk usage                                               
  - INFO: Log path: /home/michael/.cache/nvim/lsp.log                                                                          
  - INFO: Log size: 14 KB       
 ```

Closes #15567

### CODE REVIEW

File: runtime/lua/vim/lsp/health.lua
Comment (justinmk) [inline runtime/lua/vim/lsp/health.lua:17]:
```suggestion
  report_info(string.format("Log path: %s", log_path))
```

File: runtime/lua/vim/lsp/health.lua
Comment (justinmk) [inline runtime/lua/vim/lsp/health.lua:21]: Would suggest making this a warning if greater than 100MB.

```suggestion
  report_info(string.format("Log size: %d KB", log_size / 1000 ))
```

File: runtime/lua/vim/lsp/log.lua
Comment (justinmk) [inline runtime/lua/vim/lsp/log.lua:44]: indentation looks weird.

> I went ahead and added a runtime check, but we can remove it (I personally would)

why remove it? it is likely unknown to many users, and unwanted to most of them.

File: runtime/lua/vim/lsp/log.lua
Comment (justinmk) [inline runtime/lua/vim/lsp/log.lua:41]: if the variable part of a message must go in the middle of a message, try to put the longest part (filename in this case) at the end. 

```suggestion
			"LSP client log is big (%d MB): %s",
			log_info.size / (1000 * 1000),
			logfilename,
```

File: runtime/lua/vim/lsp/log.lua
Comment (mjlbach) [reply runtime/lua/vim/lsp/log.lua:44]: The runtime check? I just don't like adding things that are done automatically during the startup process. It's just an fstat, so probably not a big deal.

File: runtime/lua/vim/lsp/health.lua
Comment (mjlbach) [reply runtime/lua/vim/lsp/health.lua:13]: I wanted to emphasize that by default we don't use a log level that would cause an issue with diskspace, and that users should remove the line from their config. I can change this.

**Comment by mjlbach** [comment]
I'll clean up the commit history once the latest changes are signed off on.

File: runtime/lua/vim/lsp/log.lua
Comment (justinmk) [inline runtime/lua/vim/lsp/log.lua:104]: 2-space indnet
```suggestion
  return current_log_level
```

File: runtime/lua/vim/lsp/log.lua
Comment (justinmk) [reply runtime/lua/vim/lsp/log.lua:44]: definitely a good instinct, but this seems important enough (potentially catastrophic) to warrant some sort of active check. Doing it on (lazy) startup is a one-time cost so not too bad.

**Review by justinmk — APPROVED** [review]
LGTM after the indentation is fixed

File: runtime/lua/vim/lsp/log.lua
Comment (justinmk) [inline runtime/lua/vim/lsp/log.lua:40]: indentation :O

Comment (mjlbach) [reply runtime/lua/vim/lsp/log.lua:40]: Oh, I thought it was the indentation for which I accepted your GH edit, my bad. I'll fix.

Comment (mjlbach) [reply runtime/lua/vim/lsp/log.lua:40]: looking at this I am unclear which part of the indentation you are referring to

Comment (justinmk) [reply runtime/lua/vim/lsp/log.lua:40]: we use 2-space indent everywhere, and also the entire block starts at the wrong column

**Pull request merged**
